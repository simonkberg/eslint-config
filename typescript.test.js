/* eslint-env jest */

'use strict'

const fs = require('fs')
const path = require('path')
const config = require('./typescript.js')
const { CLIEngine } = require('eslint')

const cli = new CLIEngine({
  ignore: false,
  useEslintrc: false,
  configFile: path.resolve(__dirname, 'typescript.js'),
})

const testFile = filepath =>
  new Promise((resolve, reject) => {
    fs.readFile(filepath, 'utf-8', (err, file) => {
      if (err) {
        reject(err)
      } else {
        resolve(cli.executeOnText(file, filepath))
      }
    })
  })

expect.addSnapshotSerializer({
  print: val => JSON.stringify(val.replace(__dirname, '<PROJECT_ROOT>')),
  test: val => typeof val === 'string',
})

describe('@simonkberg/eslint-config/typescript', () => {
  it('should match the snapshot', () => expect(config).toMatchSnapshot())

  it('should load config in eslint', async () => {
    const result1 = await testFile(
      path.resolve(__dirname, '__fixtures__/typescript.ts')
    )

    expect(result1.errorCount).toBe(0)
    expect(result1.warningCount).toBe(0)
  })

  it('should support jest environment', async () => {
    const result1 = await testFile(
      path.resolve(__dirname, '__fixtures__/__tests__/typescript.ts')
    )

    expect(result1.errorCount).toBe(0)
    expect(result1.warningCount).toBe(0)

    const result2 = await testFile(
      path.resolve(__dirname, '__fixtures__/__tests__/typescript.tsx')
    )

    expect(result2.errorCount).toBe(0)
    expect(result2.warningCount).toBe(0)

    const result3 = await testFile(
      path.resolve(__dirname, '__fixtures__/typescript.test.ts')
    )

    expect(result3.errorCount).toBe(0)
    expect(result3.warningCount).toBe(0)

    const result4 = await testFile(
      path.resolve(__dirname, '__fixtures__/typescript.test.tsx')
    )

    expect(result4.errorCount).toBe(0)
    expect(result4.warningCount).toBe(0)
  })
})
