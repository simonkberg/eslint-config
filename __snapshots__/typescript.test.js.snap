// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@simonkberg/eslint-config/typescript should match the snapshot 1`] = `
Object {
  "extends": Array [
    "plugin:@typescript-eslint/recommended",
    "<PROJECT_ROOT>/node_modules/eslint-config-prettier/@typescript-eslint.js",
  ],
  "overrides": Array [
    Object {
      "files": Array [
        "*.ts?(x)",
      ],
      "rules": Object {
        "no-const-assign": "off",
        "no-dupe-args": "off",
        "no-dupe-keys": "off",
        "no-new-symbol": "off",
        "no-this-before-super": "off",
        "no-unreachable": "off",
        "valid-typeof": "off",
      },
    },
    Object {
      "env": Object {
        "jest": true,
      },
      "files": Array [
        "**/__tests__/**/*.ts?(x)",
        "**/?(*.)+(spec|test).ts?(x)",
      ],
    },
  ],
  "parser": "<PROJECT_ROOT>/node_modules/@typescript-eslint/parser/dist/parser.js",
  "plugins": Array [
    "@typescript-eslint",
  ],
  "rules": Object {
    "@typescript-eslint/camelcase": Array [
      "error",
      Object {
        "ignoreDestructuring": false,
        "properties": "never",
      },
    ],
    "@typescript-eslint/explicit-function-return-type": Array [
      "warn",
      Object {
        "allowExpressions": true,
        "allowTypedFunctionExpressions": true,
      },
    ],
    "@typescript-eslint/no-unused-vars": Array [
      "error",
      Object {
        "args": "none",
        "ignoreRestSiblings": true,
        "vars": "all",
      },
    ],
    "@typescript-eslint/no-use-before-define": Array [
      "error",
      Object {
        "classes": false,
        "functions": false,
        "typedefs": false,
        "variables": false,
      },
    ],
    "@typescript-eslint/prefer-interface": "off",
  },
}
`;
